### ISSUE TITLE
Setting SpeechServiceResponse_RequestSnr does not give SNR in detailed results json

### ISSUE THREAD

**Issue by dubajj** [Azure-Samples/cognitive-services-speech-sdk#1519]
**Describe the bug**
From the c++ sdk, I am setting RequestSNR but never receiving SNR values back.

**To Reproduce**
1. Create a SpeechConfig with the RequestSNR property: 
```         
        auto speechConfig = SpeechConfig::FromSubscription(key, region);
        speechConfig->SetSpeechRecognitionLanguage("en-US");
        speechConfig->SetOutputFormat(OutputFormat::Detailed);
        speechConfig->SetServiceProperty("SpeechServiceResponse_RequestSnr", TrueString, ServicePropertyChannel::UriQueryParameter);
        speechConfig->SetProperty(PropertyId::Speech_LogFilename, "c:\\junkyard\\speechlog.txt");
```
3.  In either the `Recognizing` or `Recognized` signal from  SpeechRecognizer, print the response json results `e.Result->Properties.GetProperty(PropertyId::SpeechServiceResponse_JsonResult);`
4. I get great output of the fragment, offset, duration and some per word confidence scores, but no SNR value appears.

**Expected behavior**
Somehow I should be able to read the current SNR value for 'recognizing' and maybe an overall one for the finished "Recognized" text. The comment for this field may be misleading me? `A boolean value specifying whether to include SNR (signal to noise ratio) in the response result. Added in version 1.18.0`

**Version of the Cognitive Services Speech SDK**
v1.21.0

**Platform, Operating System, and Programming Language**
 - OS: Windows 10, also reproduces on Ubuntu linux
 - Hardware - x64
 - Programming language: C++
 - Browser: N/A

**Additional context**

[speechlog.txt](https://github.com/Azure-Samples/cognitive-services-speech-sdk/files/8849905/speechlog.txt)
Cheers
!

**Comment by yulin-li** [comment]
Hi dubajj, I am closing this issue as internal implementation is completed and will be released with 1.23.0

**Comment by dubajj** [comment]
thank you!!
